ci: add separate coverage job

themilchenko opened the pull request:
Running tests, collecting the coverage and sending it to the coveralls was in a single job. After the patch tests, whether ran with and without coverage, were separated.

Closes #22

themilchenko committed:
ci: move tarantool installation to separate action

It's more convenient to have a separate action that installs the
necessary Tarantool. After the patch the code that installs Tarantool
was moved to a separate action.

themilchenko committed:
ci: add separate coverage job

Running tests, collecting the coverage and sending it to the coveralls
was in a single job. After the patch tests, whether ran with and without
coverage, were separated.

Closes #22

DifferentialOrange commented on .github/actions/install-tarantool/action.yml:
```diff
@@ -0,0 +1,62 @@
+runs:
+  using: "composite"
+  steps:
+    - name: Install tarantool ${{ inputs.tarantool }}
```
```suggestion
    - name: Install tarantool ${{ inputs.tarantool }} (dymanic)
```

themilchenko replied:
Edited.

DifferentialOrange commented on .github/actions/install-tarantool/action.yml:
```diff
@@ -0,0 +1,62 @@
+        make
+        make DESTDIR=${TNT_DEBUG_PATH} install
+
+    - name: Install tarantool ${{ inputs.tarantool }}
```
```suggestion
    - name: Install tarantool ${{ inputs.tarantool }} (debug)
```

themilchenko replied:
Edited.

DifferentialOrange commented on .github/workflows/testing.yaml:
```diff
@@ -21,87 +21,84 @@ jobs:
+      - name: Install Tarantool
+        uses: ./.github/actions/install-tarantool
+        with: 
+          tarantool: '3.1'
```
Since you're already reworking the CI, you can bump to 3.3 as well.

themilchenko replied:
Bumped.

oleg-jukovec replied:
Please, add a lowest supported version to a test matrix to make sure that we support Tarantool since it.

As example, the matrix could be:

[3.0.3, 3.3.1] - min and max known supported version + `master`.

themilchenko replied:
Added minimal supported and 3.3.1 versions, thank you!

themilchenko replied:
> Please, add a lowest supported version to a test matrix to make sure that we support Tarantool since it.
> 
> As example, the matrix could be:
> 
> [3.0.3, 3.3.1] - min and max known supported version + `master`.

Why is `3.0.3` is a minimal supported version? Why not 3.0.0 for example?

DifferentialOrange replied:
It seems that commits are mixed right now (`ci: add separate coverage job` still has old code)

themilchenko replied:
It's situated in the next commit: [ci: add the 3.3.1 tarantool verison](https://github.com/tarantool/metrics-export-role/pull/31/commits/e97d315716cdfd36a48ef8ddeee73ebc896ef6ff).

DifferentialOrange commented on .github/workflows/testing.yaml:
```diff
@@ -21,87 +21,84 @@ jobs:
         with:
           path: .rocks/
-          key: "cache-rocks-${{ matrix.tarantool }}${{ env.VERSION_POSTFIX }}"
+          key: "cache-rocks-3.1${{ env.VERSION_POSTFIX }}"
```
I don't get it. None of the previous steps had installed any rocks. (And even if they are, are we really want to fixate rocks based on Tarantool binary version?)

themilchenko replied:
Delete tarantool version from the `cache` key and reworked `cache` step - moved it after rocks installation.

oleg-jukovec replied:
Do we really need cache here at all? It should not take too much time to re-build rocks here.

themilchenko replied:
Agree, deleted.

themilchenko committed:
ci: add the 3.3.1 tarantool verison

Added to the test matrix Tarantool 3.3.1 version with the lowest
supported version with wich role is working.

themilchenko committed:
ci: delete rocks caching

Since the installation of rocks takes not much time, so its caching can
be removed.

DerekBum commented:
Couldn't last two commits be squashed with the `ci: add separate coverage job` one?

themilchenko commented:
> Couldn't last two commits be squashed with the `ci: add separate coverage job` one?

I think there are two logical commits: first one extends test matrix and the second one deletes rocks caching in two jobs. As for me, non of these last commits can be related  to a `ci: add separate coverage job` logic.
